master/buildbot/www/oauth2.py

```diff
@@ -197,14 +196,29 @@ class GitHubAuth(OAuth2Auth):
     tokenUri = 'https://github.com/login/oauth/access_token'
     resourceEndpoint = 'https://api.github.com'
 
+    def __init__(self,
+                 clientId, clientSecret, serverURL=None, autologin=False, **kwargs):
+
+        OAuth2Auth.__init__(self, clientId, clientSecret, autologin, **kwargs)
+        if serverURL is not None:
+            # setup for enterprise github
+            if not serverURL.endswith("/"):
+                serverURL = serverURL + "/"
+
+            GitHubAuth.authUri = '{0}login/oauth/authorize'.format(serverURL)
+            GitHubAuth.tokenUri = '{0}login/oauth/access_token'.format(serverURL)
+            GitHubAuth.resourceEndpoint = '{0}api/v3/'.format(serverURL)
+
     def getUserInfoFromOAuthClient(self, c):
-        user = self.get(c, '/user')
-        emails = self.get(c, '/user/emails')
+        map_data = c.get(self.resourceEndpoint).json()
```

Comment:
Let me double check this.. I believe had a dump and the URL was from for enterprise git, So I had changed it to use the mapping as it was intended. I will try to do this tomorrow and get an update